python/xoscar/_utils.pyx

```diff
@@ -66,13 +66,21 @@ cdef class TypeDispatcher:
     cdef _reload_lazy_handlers(self):
         for k, v in self._lazy_handlers.items():
             mod_name, obj_name = k.rsplit('.', 1)
-            with warnings.catch_warnings():
-                # the lazy imported cudf will warn no device found,
-                # when we set visible device to -1 for CPU processes,
-                # ignore the warning to not distract users
-                warnings.simplefilter("ignore")
-                mod = importlib.import_module(mod_name, __name__)
-            self.register(getattr(mod, obj_name), v)
+            imported = False
+            try:
+                with warnings.catch_warnings():
+                    # the lazy imported cudf will warn no device found,
+                    # when we set visible device to -1 for CPU processes,
+                    # ignore the warning to not distract users
+                    warnings.simplefilter("ignore")
+                    mod = importlib.import_module(mod_name, __name__)
+                imported = True
+            except Exception as e:  # pragma: no cover
```

Comment:
Can you except a `ImportError` instead of `Exception`?

Reply:
> Can you except an `ImportError` instead of `Exception`?

Add a comment here. When a python env contains cuda, but a cpu-only machine uses this env, a ``CudaAPIError`` would be raised. So here all the exceptions are caught.